### PR TITLE
aws-parallelcluster: add v3.0.0

### PR DESCRIPTION
**Pull request by hanwen-pcluste**

### CODE REVIEW

**Review by charlesg3 — DISMISSED** [review]

**Review by lukeseawalker — APPROVED** [review]

**Review by lukeseawalker — APPROVED** [review]

**Review by lukeseawalker — APPROVED** [review]

**Review by demartinofra — APPROVED** [review]

**Review by lukeseawalker — APPROVED** [review]

**Review by demartinofra — APPROVED** [review]

File: var/spack/repos/builtin/packages/aws-parallelcluster/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: According to the `setup.py`, shouldn't it be `py-tabulate@0.8.0:0.999`? Also, I see a ton of missing dependencies:
```python
    "marshmallow~=3.10",                                                                 
    "aws-cdk.core~=" + CDK_VERSION,                                                      
    "aws-cdk.aws-batch~=" + CDK_VERSION,                                                 
    "aws_cdk.aws-cloudwatch~=" + CDK_VERSION,                                            
    "aws-cdk.aws-codebuild~=" + CDK_VERSION,                                             
    "aws-cdk.aws-dynamodb~=" + CDK_VERSION,                                              
    "aws-cdk.aws-ec2~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-efs~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-events~=" + CDK_VERSION,                                                
    "aws-cdk.aws-fsx~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-imagebuilder~=" + CDK_VERSION,                                          
    "aws-cdk.aws-iam~=" + CDK_VERSION,                                                   
    "aws_cdk.aws-lambda~=" + CDK_VERSION,                                                
    "aws-cdk.aws-logs~=" + CDK_VERSION,                                                  
    "aws-cdk.aws-route53~=" + CDK_VERSION,                                               
    "aws-cdk.aws-ssm~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-sqs~=" + CDK_VERSION,                                                   
    "aws-cdk.aws-cloudformation~=" + CDK_VERSION,      
    "werkzeug~=2.0",                                                                     
    # pin connexion to 2.7.0 since 2.8.0,2.9.0 have introduced stricter requirements for werkzeug and flask.
    # This does not impact any functionality. Also next connexion release will lift this limitation
    # and we will be able to upgrade.                                                    
    "connexion==2.7.0",                                                                  
    "flask~=2.0",               
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/aws-parallelcluster/package.py
Comment (hanwen-pcluste) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: Hello Adam, 
I don't see marshmallow, aws-cdk, flask 2.0, etc are available on Spack. Should I include them even if they are not available on Spack?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: You'll need to write packages for these dependencies, otherwise you won't be able to install the package.

Comment (hanwen-pcluste) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: Is this the list of package on Spack https://spack.readthedocs.io/en/latest/package_list.html? I see the python packages have a prefix `py-`. And there is not marshmallow, aws-cdk, flask 2.0. Should I include them even if they are not available on Spack?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: Yes, that is the list of packages. You can also get this from the command-line by running `spack list`. You need to add packages for mashmallow, aws-cdk, flask, etc. if you want to install aws-parallelcluster.

Comment (hanwen-pcluste) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: So adding `depends_on` in this file is good, or we have to talk to our dependencies to let them upload their packages on Spack?

Comment (hanwen-pcluste) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: To install aws-parallelcluster from Spack, do all the dependencies have to be installed from Spack, or some of them could be installed ahead by users from pip?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: Someone (either you or the developers of those packages) needs to add build recipes for your dependencies. Until then, this PR cannot be merged.

Comment (hanwen-pcluste) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: Thank you! To make sure I understood, we need to first add the build recipes for those absent packages and then add `depends_on` for aws-parallelcluster to those dependencies?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/aws-parallelcluster/package.py:58]: You can do everything in this PR or you can submit separate PRs for those dependencies and wait until they get merged, either is fine.

**Comment by adamjstewart** [comment]
@hanwen-pcluste have you gotten a chance to add these dependencies?

**Comment by hanwen-pcluste** [comment]
Hi Adam,

Sorry for the late reply.
We decided to not release ParallelCluster 3 on Spack until all of our dependencies support Spack. ParallelCluster 2.x will still be available through Spack.

Thank you

**Comment by adamjstewart** [comment]
Since most of your dependencies are internal AWS libraries, is there anyone else at AWS who is willing to contribute them to Spack? I imagine they should be pretty easy to add. I once added all Azure ML libs and that involved hundreds of packages.

**Comment by hanwen-pcluste** [comment]
We did ask AWS CDK team. They had no plan of releasing on Spack

**Pull request closed without merging**
